storage/Devices/Disk.cc

```diff
@@ -133,7 +133,7 @@ namespace storage
     {
 	Disk* disk = dynamic_cast<Disk*>(device);
 	if (!disk)
-	    throw DeviceHasWrongType("device is not a Disk");
+	    ST_THROW(DeviceHasWrongType("device is not a Disk"));
```

Comment:
How about a dedicated check macro in Exception.h for this situation that seems to appear a lot:

#define ST_CHECK_TYPE(VAR, NAME)  if (!(VAR)) ST_THROW(DeviceHasWrongType("device is not a " #NAME))